src/runtime/contrib/thrust/thrust.cu

```diff
@@ -65,6 +65,105 @@ void thrust_sort(DLTensor* input,
+  DLTensor* values_out = args[2];
+  DLTensor* indices_out = args[3];
+  bool is_ascend = args[4];
+
+  auto data_dtype = DLDataType2String(input->dtype);
+  auto out_dtype = DLDataType2String(indices_out->dtype);
+
+  if (data_dtype == "float32") {
+    if (out_dtype == "int32") {
+      thrust_sort_nms<float, int32_t>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "int64") {
+      thrust_sort_nms<float, int64_t>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "float32") {
+      thrust_sort_nms<float, float>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "float64") {
+      thrust_sort_nms<float, double>(input, valid_count, values_out, indices_out, is_ascend);
+    } else {
+      LOG(FATAL) << "Unsupported output dtype: " << out_dtype;
+    }
+  } else if (data_dtype == "float64") {
+    if (out_dtype == "int32") {
+      thrust_sort_nms<double, int32_t>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "int64") {
+      thrust_sort_nms<double, int64_t>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "float32") {
+      thrust_sort_nms<double, float>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "float64") {
+      thrust_sort_nms<double, double>(input, valid_count, values_out, indices_out, is_ascend);
+    } else {
+      LOG(FATAL) << "Unsupported output dtype: " << out_dtype;
+    }
+  } else if (data_dtype == "int32") {
+    if (out_dtype == "int32") {
+      thrust_sort_nms<int32_t, int32_t>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "int64") {
+      thrust_sort_nms<int32_t, int64_t>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "float32") {
+      thrust_sort_nms<int32_t, float>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "float64") {
+      thrust_sort_nms<int32_t, double>(input, valid_count, values_out, indices_out, is_ascend);
+    } else {
+      LOG(FATAL) << "Unsupported output dtype: " << out_dtype;
+    }
+  }  else if (data_dtype == "int64") {
+    if (out_dtype == "int32") {
+      thrust_sort_nms<int64_t, int32_t>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "int64") {
+      thrust_sort_nms<int64_t, int64_t>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "float32") {
+      thrust_sort_nms<int64_t, float>(input, valid_count, values_out, indices_out, is_ascend);
+    } else if (out_dtype == "float64") {
+      thrust_sort_nms<int64_t, double>(input, valid_count, values_out, indices_out, is_ascend);
+    } else {
+      LOG(FATAL) << "Unsupported output dtype: " << out_dtype;
+    }
+  } else {
+    LOG(FATAL) << "Unsupported input dtype: " << data_dtype;
+  }
+});
+
```

Comment:
Likely we can share this if/else with `thrust_sort` using template or lambda

Reply:
Could you elaborate more on this?

Reply:
I mean we shouldn't duplicate this if/else blocks. See my comment below.